test/diagonalGate.jl

```diff
@@ -154,3 +154,93 @@ end
    
 end
 
+@testset "Diagonal Gate: N targets" begin
+
+
+    ######################################################
+
+    diagonalGate_N_targets(target_list::Vector{<:Integer}) = 
+        Diagonal_N_TargetGate(target_list)
+
+    struct Diagonal_N_TargetGate <: Snowflake.AbstractGate
```

Comment:
If there are no multi-qubit diagonal gates, rather remove the implementation than define a gate here.